torch/onnx/symbolic_helper.py

```diff
@@ -800,6 +800,35 @@ def _interpolate_get_scales_and_mode(g, input, size, scale_factor, mode, align_c
     return scale_factor, mode
 
 
+def _argmin_argmax_helper(g, input, dim, keepdim, op_name):
+    keepdim = _parse_arg(keepdim, "i")
+
+    def op_wrapper(input, axis_i, keepdims_i):
+        if GLOBALS.export_onnx_opset_version >= 12:
+            return g.op(
+                op_name,
+                input,
+                axis_i=axis_i,
+                keepdims_i=keepdims_i,
+                select_last_index_i=False,
+            )
+        return g.op(op_name, input, axis_i=axis_i, keepdims_i=keepdims_i)
+
+    if _is_none(dim):
+        flattened = _reshape_helper(
+            g, input, g.op("Constant", value_t=torch.tensor([-1]))
+        )
+        output = op_wrapper(flattened, axis_i=0, keepdims_i=False)
+        if keepdim:
+            input_shape = g.op("Shape", input)
+            new_shape = g.op("Div", input_shape, input_shape)
```

Comment:
I see that we are creating a `ones(n)` tensor here. Is that accurate? @qqaatw If so do we think there should be a more straightforward representation for it? @BowenBao

Reply:
It's int64-int64 division so pretty accurate I think. I have scrutinized it but didn't come up any more concise and straightforward approach.

Reply:
Do you have a better solution?

Reply:
Does `ones_like` work?

Reply:
Does ConstantOfShape work?

Reply:
1. Where does the shape come from?
2. It is a static approach I think.

Reply:
What I was thinking was something like

```python
input_shape = g.op("Shape", input)
input_shape_shape = g.op("Shape", input_shape)
new_shape = g.op("ConstantOfShape", input_shape_shape, value_t=torch.tensor( 
    [1], dtype=torch.int64))
```

maybe i was missing something. Can you saw more how it is static?

Reply:
Thanks for providing the snippet! I think it's feasible and able to handle tensors dynamically.

However, in terms of straightforward, I think your approach is more complex than the current one because IMO it's harder to know the intention of creating  an `ones` tensor without giving a comment.

In addition, though I'm not familiar with the graph implementation of ORT, in general sense your approach may result in more resource usage:

```python
input_shape = g.op("Shape", input) # 1st memory alloc op
input_shape_shape = g.op("Shape", input_shape) # 2nd memory alloc op
new_shape = g.op("ConstantOfShape", input_shape_shape, value_t=torch.tensor( 
    [1], dtype=torch.int64)) # 3rd memory alloc op
```

While the current approach:
```
input_shape = g.op("Shape", input)  # 1st memory alloc op
new_shape = g.op("Div", input_shape, input_shape)  # arithmetic op (if the storage not in-place then 2nd memory alloc)
```

Let me know if I'm missing something.

Reply:
Thanks for sharing the thought process. I will get more opinions and update here

Reply:
In principle we prioritize clarity in the conversion results, with the assumption that optimization can be done outside of the conversion process. When we see `Div` we expect some values to be _computed_. In this case it is used for getting a bunch of ones of a certain shape, so as to maintain the rank in the result, which is not really an arithmetic problem (and thus may cause confusion).

An additional inline comment is definitely helpful and does no harm.

Reply:
ok.